src/main/java/io/adobe/cloudmanager/Environment.java

```diff
@@ -89,19 +140,87 @@ public interface Environment {
    *
    * @param variables the variables to set
    * @return the complete list of variables in this environment
-   * @throws CloudManagerApiException when any error occurs.
+   * @throws CloudManagerApiException when any error occurs
    */
   Set<Variable> setVariables(Variable... variables) throws CloudManagerApiException;
 
+  /**
+   * Resets this environment, if it is of type RDE.
+   *
+   * @throws CloudManagerApiException when any error occurs;
+   */
+  void reset() throws CloudManagerApiException;
+
+  /**
+   * Retrieve the Developer Console URL for this Environment.
+   *
+   * @return the url to the developer console.
+   * @throws CloudManagerApiException when any error occurs
+   */
+  String getDeveloperConsoleUrl() throws CloudManagerApiException;
+
   /**
    * Downloads the logs for this environment
    *
    * @param logOption the log file reference
    * @param days      the number of days to download
    * @param dir       the directory in which to place the log files
    * @return a list of EnvironmentLogs with details about the downloaded files
-   * @throws CloudManagerApiException when any error occurs.
+   * @throws CloudManagerApiException when any error occurs
    */
   Collection<EnvironmentLog> downloadLogs(LogOption logOption, int days, File dir) throws CloudManagerApiException;
 
+  /**
+   * Representation of the different environment types.
+   */
+  enum Type {
+    DEV,
+    STAGE,
+    PROD,
+    RDE
+  }
+
+  @Getter
+  enum Tier {
+    AUTHOR("author"),
```

Comment:
Since the value is the same as the enum constant name, couldn't you just do a case insensitive comparison? An enum has its name as a method already doesn't it?

Reply:
I really should just make these not use the underlying impl. So the consumers don't have to care about the literal values. Take care of dong transformations in the details away from consumer's knowledge.